src/model/jwks.rs

```diff
@@ -164,15 +179,56 @@ impl Jwk {
     }
 }
 
+fn generate_x509_cert(key_pair: &PKey<Private>) -> Result<X509, ErrorStack> {
+    let mut x509_name = X509NameBuilder::new()?;
+    x509_name.append_entry_by_text("C", "IT")?;
+    x509_name.append_entry_by_text("O", "Prima CA")?;
+    x509_name.append_entry_by_text("CN", "Prima CA")?;
+    let x509_name = x509_name.build();
+
+    let mut cert_builder = X509::builder()?;
+    cert_builder.set_version(2)?;
+    let serial_number = {
+        let mut serial = BigNum::new()?;
+        serial.rand(159, MsbOption::MAYBE_ZERO, false)?;
+        serial.to_asn1_integer()?
+    };
+    cert_builder.set_serial_number(&serial_number)?;
+    cert_builder.set_subject_name(&x509_name)?;
+    cert_builder.set_issuer_name(&x509_name)?;
+    cert_builder.set_pubkey(key_pair)?;
+    let not_before = Asn1Time::days_from_now(0)?;
+    cert_builder.set_not_before(&not_before)?;
+    let not_after = Asn1Time::days_from_now(365)?;
+    cert_builder.set_not_after(&not_after)?;
+
+    cert_builder.append_extension(BasicConstraints::new().critical().ca().build()?)?;
+    cert_builder.append_extension(KeyUsage::new().critical().key_cert_sign().crl_sign().build()?)?;
+
+    let subject_key_identifier = SubjectKeyIdentifier::new().build(&cert_builder.x509v3_context(None, None))?;
+    cert_builder.append_extension(subject_key_identifier)?;
+
+    cert_builder.sign(key_pair, MessageDigest::sha256())?;
+    let cert = cert_builder.build();
+
+    Ok(cert)
+}
+
 #[cfg(test)]
 mod tests {
     use crate::error::Error;
+    use crate::model::ca::CA;
```

Comment:
The CA struct was removed in https://github.com/primait/localauth0/pull/25/commits/d4334885207427d29f784e2027a5627bba4ea9d8, which means this test doesn't compile

We really should run tests as a part of our CI

Reply:
Thanks, fixed!